### PR TITLE
Change position using drag and drop

### LINKED ISSUE #70: Change the position of books within the state via drag/drop
### Overview

A long press on the items in the main page should allow a drag-and-drop gesture. Each book can be repositioned to another position. 

### Implementation

If there is any other sorting strategy chosen than `default`, the default sorting strategy should be selected after updating the positions.

### PR DESCRIPTION
**Pull request by lockieRichter**
Also fixes broken google sign in.

### CODE REVIEW

**Comment by lockieRichter** [comment]
Not sure why web build is failing in GitHub actions, web seems to be building fine for me locally

**Comment by lockieRichter** [comment]
Also, @shockbytes is the idea to implement this for Web too?

File: lib/src/providers/authentication.dart [hunk 1]
```diff
@@ -42,13 +43,13 @@ Future<DanteUser?> user(UserRef ref) {
...
-@riverpod
+@Riverpod(keepAlive: true)
 GoogleSignIn googleSignIn(
   GoogleSignInRef ref,
 ) =>
     GoogleSignIn(
-      clientId:
-          '150599422814-moto7djse1tf7vtso7slemniki76ohg6.apps.googleusercontent.com',
+      // clientId:
+      //     '150599422814-moto7djse1tf7vtso7slemniki76ohg6.apps.googleusercontent.com',
```
Comment (shockbytes) [inline lib/src/providers/authentication.dart]: Why is this necessary?

**Review by shockbytes — APPROVED** [review]

**Comment by shockbytes** [comment]
@lockieRichter I approved it. 

It does not need to be implemented on Web for the moment. 

The build breaks. You need to update the dependency `web` to version `>= 0.4.2` to make it build.

File: lib/src/providers/authentication.dart (hunk 1 above)
Comment (lockieRichter) [reply lib/src/providers/authentication.dart]: oh yeah, sorry meant to delete this. All our google sign in stuff should be handled now in the install of the package, so we don't need to actually specify this here (see https://pub.dev/packages/google_sign_in_ios#ios-integration)

**Comment by lockieRichter** [comment]
No worries @shockbytes, to fix this issue I've had to bump the minimum Dart version in pubspec to `3.3.0`, to be able to support this you'll need to use the latest flutter `3.19.2`
